### PR TITLE
Migrate version to header and fix cache warnings

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Fixes a warning in cache.c wrt null termination.

### CODE REVIEW

File: vkd3d_version.h.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#define VKD3D_BUILD_SIZE 8
+#define VKD3D_BUILD_MAX_LENGTH (VKD3D_BUILD_SIZE - 1)
+
+static const char vkd3d_build[VKD3D_BUILD_SIZE] = "@VCS_TAG@";
```
Comment (doitsujin) [inline vkd3d_version.h.in:4]: does this work even if the build tag has more than 7 characters?

**Comment by Saancreed** [comment]
I'd like to suggest an alternative (but I'm not sure it it'll work like I think it will), to use `uint64_t` to store an abbreviated commit hash instead. Like this:

```
command : ['git', 'describe', '--always', '--exclude=*', '--abbrev=16']
```
and
```c
static const uint64_t vkd3d_build = 0x@VCS_TAG@;
```

This way we still use a fixed–width revision identifier, but we are slightly more efficient with that space.

One thing to consider here would be telling apart clean and dirty working trees, but I suppose it would be possible to use `--abbrev=15` and `--dirty=0` which would shorten the identifier to 60 bits and shift them to the left by 4 when there are uncommited changes.

But I'm probably overthinking this :upside_down_face:

**Comment by doitsujin** [comment]
@Saancreed I actually quite like the idea; question is whether or not we need the old build string in other places and whether Meson allows us to essentially add two different tags.

If in doubt, we could always parse the version string though.

**Comment by HansKristian-Work** [comment]
Is this PR still relevant?

**Comment by Joshua-Ashton** [comment]
Nope.

**Pull request closed without merging**